fdbserver/CommitProxyServer.actor.cpp

```diff
@@ -282,6 +354,14 @@ ACTOR Future<Void> commitBatcher(ProxyCommitData* commitData,
 						    .detail("Size", bytes)
 						    .detail("Client", req.reply.getEndpoint().getPrimaryAddress());
 					}
+
+					if (!verifyPrefix(commitData, req)) {
+						++commitData->stats.txnCommitErrors;
+						req.reply.sendError(illegal_tenant_access());
+						TraceEvent(SevWarnAlways, "TenantPrefixMismatch").suppressFor(60);
```

Comment:
I like this trace event, maybe we could move it up into the `verifyPrefix` function and have it also include details for the tenant's prefix and the key that didn't match.